datamodels/2.x/itop-portal-base/portal/src/Form/ObjectFormManager.php

```diff
@@ -990,54 +1010,88 @@ public function Build()
 			}
 		}
 
-		// Checking if the instance has attachments
-		if (class_exists('Attachment') && class_exists('AttachmentPlugIn'))
-		{
-			// Checking if the object is allowed for attachments
-			$bClassAllowed = false;
-			$aAllowedClasses = MetaModel::GetModuleSetting('itop-attachments', 'allowed_classes', array('Ticket'));
-			foreach ($aAllowedClasses as $sAllowedClass)
+		// fallback Checking if the instance has attachments
+		// (in case attachment is not explicitly declared in layout)
+		if (!$this->IsPluginInitialized(AttachmentPlugIn::class)
+		&& AttachmentPlugIn::IsAttachmentAllowedForObject($this->oObject)){
```

Comment:
```suggestion
		if (class_exists('Attachment') && class_exists('AttachmentPlugIn')
		&& !$this->IsPluginInitialized(AttachmentPlugIn::class)
		&& AttachmentPlugIn::IsAttachmentAllowedForObject($this->oObject)){
```

We might want to check if the attachment module is present, otherwise this will crash.